### ISSUE TITLE
Support "Use this node as much as possible" build scheduling

### ISSUE THREAD

**Issue by gurchik** [jenkinsci/amazon-ecs-plugin#188]
With ec2-plugin we have an instance defined with "Usage" set to "Use this node as much as possible," and other instances defined with "Only build jobs with label expressions matching this node". With this setup, for almost all of our numerous jobs we do not set a label expression when we want it to use the default node, and only for the few jobs we want to use a specific node do we set a label expression.

It looks like the only possible behavior with amazon-ecs-plugin is for ECS nodes to be created with "Only build jobs with label expressions matching this node." That means we have to set a label expression on all of our jobs matching the ECS node we want to use by default. While this isn't a huge problem for _me_, it would make more sense if we could configure ECS nodes to "Use this node as much as possible" because this is the default behavior when you create nodes without the use of a plugin like ec2-plugin or amazon-ecs-plugin.

**Comment by DonaldChan** [comment]
Is this configurable anywhere in the plugin?

**Comment by cgrubesol** [comment]
I would like to have this issue reopened as this is an important feature for us as well. We are attempting to support 7 or so different developer communities each with countless numbers of builds. Prior to migrating to ECS, we were employing the YAD plugin which supports the "Use this node as much as possible" feature. With this option set, the YAD plugin will provision container agents for builds in which the node selector is not specified. It is a pretty big ask for us to get all of our users to explicitly select a node for all of their builds. Also it seems like having a default ECS template is useful functionality. 

I attempted not specifying a label at all in my ECS plugin config, but this doesn't appear to work. Anyone know of any other potential workarounds?

**Comment by shushry** [comment]
I agree with above - my expectation on this plugin, is that jenkins will spin up an agent on demand - and the job itself wouldn't have to be specifically configured to use this cloud agent.   I'm going to open a new ticket on this, but it looks like the plugin is looking for maintainers...
